File: bin/get-bin-names.py
#/usr/bin/env python
import os, json;

if os.environ['npm_config_global'] == 'true':
    exit(0)

package_json_path=os.environ['PWD'] + '/package.json'

# create shims only for globally installed pkg
if package_json_path.count('node_modules') > 1:
    exit(0)

with open(package_json_path, 'r') as package_json_file:
	package_json = json.load(package_json_file)

if 'bin' in package_json:
    if isinstance(package_json['bin'], dict):
        bin_list = " ".join(package_json['bin'].keys())
        print(bin_list)
    else:
        print(package_json["name"])

File: bin/get-bin-names.js
#!/usr/bin/env node

var fs = require('fs');

if (process.env['npm_config_global'] != 'true') {
  process.exit();
}

var package_json_path=process.env['PWD'] + '/package.json';
var node_modules_occurance = (package_json_path.match(/node_modules/g) || []).length;

// create shims only for globally installed pkg
if (node_modules_occurance > 1) {
  process.exit();
}

var package_json = JSON.parse(fs.readFileSync(package_json_path, 'utf8'));

if (!('bin' in package_json)) {
  process.exit();
}

if (typeof package_json['bin'] === "object") {
  var bin_names = Object.keys(package_json['bin']);
  console.log(bin_names.join(" "));
}
else
{
  console.log(package_json["name"]);
}
